### ISSUE TITLE
Amiga Forever / C64 Forever windows are not captured

### ISSUE THREAD

**Issue by mattcharlton** [obsproject/obs-studio#5397]
### Operating System Info

Windows 10

### OBS Studio Version

27.1.1

### OBS Studio Version (Other)

N/A

### OBS Studio Log URL

N/A

### OBS Studio Crash Log URL

N/A

### Expected Behavior

Gameplay to be recorded/streamed as per any other window that is being captured.

### Current Behavior

A grey screen is captured/recorded instead of the game window.  This is due to the way that the application uses child processes to pull together all of the content seen in the application by the end user.

### Steps to Reproduce

1. Set up OBS to capture a specific window
2. Choose an Amiga Forever/ C64 Forever window
3. you'll see a grey screen rather than the game you're playing.
...

### Anything else we should know?

Reported here back in 2019:

https://obsproject.com/forum/threads/amiga-forever-and-c64-forever-playback-windows.105871/

With example CPP code to allow recognition of the windows for capture.

The code is well commented and looks to be everything someone would need to fix, but c++ isn't one of the languages I'm massively familiar with and I wouldn't know where to start but hopefully there is someone out there who can adapt and add this code to fix the problem.

**Comment by MikeLabatt** [comment]
> just because a random sample of code has been provided, does not mean it will be easy to implement in to our capture functions. All this code seems to do is enumerate the child windows, which I don't think is the actual challenge for capture

If the sample code works in enumerating all windows that need to be captured, what is the actual challenge for capture? How can the app developers (who are not OBS experts) help OBS developers and users?

**Comment by jp9000** [comment]
I don't think we should expect a game to have to render a way we want in order for something to be capturable. Rendering with multiple windows/swapchains should be perfectly acceptable, even if it's an uncommon thing to do.

That being said, choosing a specific window/swapchain to capture within a single program has its own problems. A fair number of things would need to be changed internally in order to make this feasible. This unfortunately was not quite how game capture was designed, so the bad news is I wouldn't expect a fix for this any time soon.

**Comment by jitspoe** [comment]
I'm running into a similar issue.  I have an overlay + twitch chat viewer that I'm working on in the Godot engine.  In Godot 4.2, everything was working properly, but in the newer versions of Godot, something must have changed with the way windows were initialized, and now, no matter what I set the window to in "Capture specific window", it always captures the chat window, not the overlay.  I'm guessing OBS just grabs the first (or maybe most recent?) hardware accelerated instance on the process rather than what's associated with the window?

Not sure if it's clear from the screenshot here:

![Image](https://github.com/user-attachments/assets/16b9a1ae-8479-4aee-ad05-e7b8b2fc0513)

I have the "Wulfmanbot (DEBUG)" window selected, but it's displaying the contents of the "Twitch Chat" window.  If I use a regular window capture, the correct window is captured, however transparency only works with the game capture, so I need to figure out how to make Godot initialize the windows in an order so that the overlay is what's captured or OBS needs to capture the contents of the window specified, somehow.

**Comment by Tiraelina** [comment]
Much like the above Godot example. This is also a problem in Unreal when trying to create additional windows and just the Unreal Editor itself. When I add a new SWindow, game capture will instantly capture it instead until it's closed or minimized with another window visible.
